File: backend/service-cart-spring/src/main/java/com/cart/servicecartspring/v1/repository/CartDetailRepository.java
package com.cart.servicecartspring.v1.repository;

import com.cart.servicecartspring.v1.domain.CartDetail;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface CartDetailRepository extends JpaRepository<CartDetail, Integer> {
}

File: backend/service-cart/v1/repository/cart_detail_repository.go
package repository

import (
	"github.com/google/uuid"
	"gorm.io/gorm"
	"service-cart/internal/domain"
)

type CartDetailRepositoryImpl struct {
	db *gorm.DB
}

func (c CartDetailRepositoryImpl) InsertCartDetail(detail domain.CartDetail) error {
	//TODO implement me
	panic("implement me")
}

func (c CartDetailRepositoryImpl) UpdateCartDetail(detail domain.CartDetail) error {
	//TODO implement me
	panic("implement me")
}

func (c CartDetailRepositoryImpl) FindByCartIdAndProductId(cartId uuid.UUID, productId int) *domain.CartDetail {
	//TODO implement me
	panic("implement me")
}

func NewCartDetailRepository(db *gorm.DB) domain.CartDetailRepository {
	return &CartDetailRepositoryImpl{
		db: db,
	}
}
